packages/loopring_v3/contracts/amm/libamm/AmmWithdrawal.sol

```diff
@@ -32,6 +34,30 @@ library AmmWithdrawal
         S.exchange.withdrawFromApprovedWithdrawals(owners, tokens);
     }
 
+    // This is to claim other people depositing into this AMM pool.
+    function withdrawFromDepositRequests(
+        AmmData.State storage  S,
+        address[]     calldata tokens
+        )
+        internal
+    {
+        require(tokens.length > 0, "INVALID_TOKENS");
+
+        address recipient = S.exchange.owner();
+
+        for (uint i = 0; i < tokens.length; i++) {
+            address token = tokens[i];
+            S.exchange.withdrawFromDepositRequest(address(this), token);
+            if (token != address(this)) {
+                uint balance = token == address(0) ?
+                    address(this).balance :
+                    ERC20(token).balanceOf(address(this));
+
+                AmmUtil.transferOut(token, balance, recipient);
+            }
```

Comment:
Seems like I missed this previously, but because this will transfer out the full balance (not just what was withdrawn from the deposit) we have to implement the same logic here as in `canDrain`, we cannot allow this for tokens of the pool when the pool is offline.

So perhaps we should just call `canDrain` here, or similar to `withdrawFromApprovedWithdrawals` don't actually transfer any tokens out here, and leave that to `drain`.

Reply:
Brecht, I think we can delete this function and `withdrawFromApprovedWithdrawals` and the Drainable interface. Because in all our expected use cases, there will be no onchain deposit or withdrawals. What do you think?

Reply:
Okay for me, they would indeed only be useful in case people would use the pool incorrectly. And `withdrawFromApprovedWithdrawals` is just a helper function.